services/user/Webmail/ui/src/hooks/use-mail.ts

```diff
@@ -0,0 +1,131 @@
+import { atom, useAtom } from "jotai";
+
+import { useUser } from "./use-user";
+import { useQuery } from "@tanstack/react-query";
+import { useLocalStorage } from "./use-local-storage";
+
+const composeAtom = atom(false);
+export function useCompose() {
+    return useAtom(composeAtom);
+}
+
+export type Message = {
+    id: string;
+    from: string;
+    to: string;
+    datetime: number;
+    status: "draft" | "sent";
+    read: boolean;
+    inReplyTo: string | null;
+    subject: string;
+    body: string;
+};
+
+type RawMessage = {
+    sender: string;
+    receiver: string;
+    subject: string;
+    body: string;
+};
+
+const transformRawMessagesToMessages = (rawMessages: RawMessage[]) => {
+    return rawMessages.reverse().map(
+        (msg, i) =>
+            ({
+                id: `${msg.sender}-${msg.receiver}-${msg.subject}-${msg.body}`,
+                from: msg.sender,
+                to: msg.receiver,
+                datetime: Date.now() - i * 1_000_000,
+                status: "sent",
+                read: false,
+                inReplyTo: null,
+                subject: msg.subject,
+                body: msg.body,
+            }) as Message,
+    );
+};
+
+const getIncomingMessages = async (account: string) => {
+    const res = await fetch(`/messages?receiver=${account}`);
+    const rawMessages = (await res.json()) as RawMessage[];
```

Comment:
Given we already have it installed it would be kinda nice to `zod.parse` this with `strict` on as well.

Reply:
Good idea. I'll add that to the fast follow in the description above.